util/fipstools/acvp/acvptool/subprocess/kdf.go

```diff
@@ -67,59 +106,114 @@ func (k *kdfPrimitive) Process(vectorSet []byte, m Transactable) (interface{}, e
 		}
 
-		if group.CounterLocation != "after fixed data" {
-			// We only support the counter location being after fixed data
-			return nil, fmt.Errorf("label location %q not supported", group.CounterLocation)
+		groupResp, err := groupProcessor(group, m)
+		if err != nil {
+			return nil, err
 		}
+		respGroups = append(respGroups, groupResp)
+	}
+
+	return respGroups, nil
+}
+
+func processFeedbackMode(group kdfTestGroup, m Transactable) (kdfTestGroupResponse, error) {
+	groupResp := kdfTestGroupResponse{ID: group.ID}
+
+	if group.OutputBits%8 != 0 {
+		return kdfTestGroupResponse{}, fmt.Errorf("%d bit key in test group %d: fractional bytes not supported", group.OutputBits, group.ID)
+	}
 
-		if group.CounterBits != 8 {
-			// We only support counter lengths of 8
-			return nil, fmt.Errorf("counter length %q not supported", group.CounterBits)
+	if group.CounterLocation != AfterFixedDataCounterLocation {
+		// We only support the counter location being after fixed data
+		return kdfTestGroupResponse{}, fmt.Errorf("label location %q not supported", group.CounterLocation)
+	}
+
+	if group.CounterBits != 8 {
+		// We only support counter lengths of 8
+		return kdfTestGroupResponse{}, fmt.Errorf("counter length %v not supported", group.CounterBits)
+	}
+
+	outputBytes := uint32le(group.OutputBits / 8)
+
+	// Fixed data variable is determined by the IUT according to the NIST specifications
+	// We send it as part of the response so that NIST can verify whether it is correct
+	fixedData := make([]byte, 4)
+	rand.Read(fixedData)
+
+	for _, test := range group.Tests {
+		test := test
+		testResp := kdfTestResponse{ID: test.ID}
+
+		// Make the call to the crypto module.
+		resp, err := m.Transact("KDF/Feedback/"+group.MACMode, 1, outputBytes, test.Key, fixedData)
+		if err != nil {
+			return kdfTestGroupResponse{}, fmt.Errorf("wrapper KDF operation failed: %s", err)
 		}
 
-		outputBytes := uint32le(group.OutputBits / 8)
+		// Parse results.
+		testResp.ID = test.ID
+		testResp.KeyOut = resp[0]
+		testResp.FixedData = fixedData
+
+		groupResp.Tests = append(groupResp.Tests, testResp)
+	}
+
+	return groupResp, nil
+}
+
+func processCounterMode(group kdfTestGroup, m Transactable) (kdfTestGroupResponse, error) {
+	groupResp := kdfTestGroupResponse{ID: group.ID}
 
-		// Fixed data variable is determined by the IUT according to the NIST specifications
-		// We send it as part of the response so that NIST can verify whether it is correct
-		fixedData := make([]byte, 4)
-		rand.Read(fixedData)
+	if group.OutputBits%8 != 0 {
+		return kdfTestGroupResponse{}, fmt.Errorf("%d bit key in test group %d: fractional bytes not supported", group.OutputBits, group.ID)
+	}
+
+	if group.CounterLocation != BeforeFixedDataCounterLocation {
+		// We only support the counter location being after fixed data
+		return kdfTestGroupResponse{}, fmt.Errorf("label location %q not supported", group.CounterLocation)
```

Comment:
Same comment/question
```suggestion
		return kdfTestGroupResponse{}, fmt.Errorf("counter location %q not supported", group.CounterLocation)
```